Content.Client/GameTicking/Managers/ClientGameTicker.cs

```diff
@@ -153,8 +162,35 @@ private void RoundEnd(RoundEndMessageEvent message)
             if (_window?.RoundId == message.RoundId)
                 return;
 
-            //This is not ideal at all, but I don't see an immediately better fit anywhere else.
-            _window = new RoundEndSummaryWindow(message.GamemodeTitle, message.RoundEndText, message.RoundDuration, message.RoundId, message.AllPlayersEndInfo, _entityManager);
+            //Cyberfinn changes:
+            //back up the round info:
+            _roundEndContainer._message = message;
+            //display the window
+            DisplayRoundEndSummary(_roundEndContainer._message);
+
+            //enable the button, so that users can review the summary after closing the window
+            EnableRoundEndSummaryButton();
+        }
+        //Cyberfinn changes: moved this out to its own method
+        public static void DisplayRoundEndSummary(RoundEndMessageEvent? message)
+        {
+            if (message != null)
+                _window = new RoundEndSummaryWindow(message.GamemodeTitle, message.RoundEndText, message.RoundDuration, message.RoundId, message.AllPlayersEndInfo, _entityManager);
+        }
+
+        private void EnableRoundEndSummaryButton()
+        {
+            //enable the button, so we can still click it once we've closed round-end window - until next round starts, where we'll disable it again
+            if (EscapeUIController._escapeWindow != null)
+                if (EscapeUIController._escapeWindow.buttonShowRoundEnd != null)
+                    EscapeUIController._escapeWindow.buttonShowRoundEnd.Visible = true;
+        }
+        private void DisableRoundEndSummaryButton()
+        {
+            //enable the button, so we can still click it once we've closed round-end window - until next round starts, where we'll disable it again
+            if (EscapeUIController._escapeWindow != null)
+                if (EscapeUIController._escapeWindow.buttonShowRoundEnd != null)
+                    EscapeUIController._escapeWindow.buttonShowRoundEnd.Visible = false;
```

Comment:
similar thing also consider just having it be a bool argument since these functions are identical

Reply:
ah I see what you mean now, good idea! Might just keep as-is tho, with the changes you suggested. I like having it in its' own method lol